### PR TITLE
Search: a few minor improvements

### PR DESCRIPTION
**Pull request by ryantxu**
This PR:

1. Implements "Recent" and "Starred" queries.  TODO: (no stars behavior)
2. Shows the panel icon in "grid" view
3. Fixes the tag query when no search term is given
4. Fix padding in folder view
5. Fix folder/item selection in "Folder" view.  
6. "type" column is now accurate for pages 2...n (it does not use the accessor anymore)

### CODE REVIEW

File: public/app/features/search/page/components/FolderSection.tsx [hunk 1]
```diff
@@ -38,15 +40,27 @@ export const FolderSection: FC<SectionHeaderProps> = ({ section, selectionToggle
...
     };
     if (section.title === 'Starred') {
-      // TODO
+      const stars = await getBackendSrv().get('api/user/stars');
+      if (stars.length > 0) {
+        query = {
+          uid: stars, // array of UIDs
+        };
+      } else {
+        // ??
```
Comment (ryantxu) [inline public/app/features/search/page/components/FolderSection.tsx:55]: What do we do when nothing is starred?

File: public/app/features/search/page/components/columns.tsx [hunk 2]
```diff
@@ -225,11 +224,11 @@ function makeTypeColumn(
   styles: Record<string, string>
 ): TableColumn {
   return {
-    Cell: DefaultCell,
     id: `column-type`,
     field: kindField ?? typeField,
     Header: 'Type',
-    accessor: (row: any, i: number) => {
```
Comment (ryantxu) [inline public/app/features/search/page/components/columns.tsx:232]: using accessor only works for the first page.  Now that we are using the `InfiniteLoader`, we need to do this all in the `Cell` implementation

File: public/app/features/search/page/components/FolderSection.tsx (hunk 1 above)
Comment (axelavargas) [reply public/app/features/search/page/components/FolderSection.tsx:57]: 🤔 I think we could keep showing the "starred" folder and we can add a message in the frontend "No dashboards are starred.",  something like:
EmptyStarred

Not sure if we have something for this pattern from UX point of view, what do you think @amy-super

**Review by axelavargas — APPROVED** [review]
It works great! 👏🏾  I left a small suggestion when we don't have "starred" dashboards.

**Pull request merged**